pkg/storage/storage.go

```diff
@@ -502,6 +504,16 @@ func (is *ImageStore) DeleteImageManifest(repo string, reference string) error {
 		return err
 	}
 
+	oci, err := umoci.OpenLayout(dir)
+	if err != nil {
+		return err
+	}
+	defer oci.Close()
+
+	if err := oci.GC(context.Background()); err != nil {
```

Comment:
With dedupe, delete should be very fast since we are removing links for most part.